std-bits/base/src/main/java/org/enso/base/enso_cloud/TransientHTTPResponseCache.java

```diff
@@ -0,0 +1,428 @@
+package org.enso.base.enso_cloud;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpHeaders;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.enso.base.net.URIWithSecrets;
+import org.graalvm.collections.Pair;
+
+/**
+ * TransientHTTPResponseCache is a cache for EnsoHttpResponse values that respects the cache control
+ * HTTP headers received in the original repsonse to a request.
+ *
+ * <p>It also puts limits on the size of files that can be requested, and on the total cache size,
+ * deleting entries to make space for new ones.
+ *
+ * <p>Without cachineg, EnsoHttpResponse contains an InputStream providing the response data. When
+ * there is a cache hit, this stream reads from the local file storing the cached data. When there
+ * is no cache hit, the InputStream is connected directly to the remote server.
+ */
+public class TransientHTTPResponseCache {
+  TransientHTTPResponseCache() {}
+
+  private static final Logger logger = Logger.getLogger(TransientHTTPResponseCache.class.getName());
+
+  // 1 year.
+  private static final int DEFAULT_TTL_SECONDS = 31536000;
```

Comment:
Is this some standardized value? If not, isn't it too large for the purposes of this cache? It feels like something around 24h would be more fitting, no?

Reply:
This was suggested by @jdunkerley .